File: arrays/longest-harmonious-subsequence/longest-harmonious-subsequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
function findLHS(nums) {
  const map = {};
  let max = 0;

  for (let i = 0; i < nums.length; i++) {
    if (!map[nums[i]]) {
      map[nums[i]] = 0;
    }

    map[nums[i]]++;

    if (map[nums[i] - 1]) {
      max = Math.max(max, map[nums[i]] + map[nums[i] - 1]);
    }

    if (map[nums[i] + 1]) {
      max = Math.max(max, map[nums[i]] + map[nums[i] + 1]);
    }
  }

  return max;
}

File: arrays/longest-harmonious-subsequence/longest-harmonious-subsequence.java
public class Solution {
  public int findLHS(int[] nums) {
    HashMap<Integer, Integer> map = new HashMap<>();
    int max = 0;
    for (int n : nums) {
      map.put(n, map.computeIfAbsent(n, k -> 0) + 1);
      if (map.containsKey(n + 1)) {
        max = Math.max(max, map.get(n) + map.get(n + 1));
      }

      if (map.containsKey(n - 1)) {
        max = Math.max(max, map.get(n) + map.get(n - 1));
      }
    }
    return max;
  }
}
